File: app/src/main/java/com/gxd/demo/design/pattern/chain/interceptor/impl/RetryAndFollowUpInterceptor.java
package com.gxd.demo.design.pattern.chain.interceptor.impl;

import com.gxd.demo.design.pattern.chain.interceptor.Chain;
import com.gxd.demo.design.pattern.chain.interceptor.Interceptor;
import com.gxd.demo.design.pattern.chain.interceptor.Request;
import com.gxd.demo.design.pattern.chain.interceptor.Response;

public class RetryAndFollowUpInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) {
        Request request = ((ChainImpl) chain).request;
        return chain.process(request);
    }
}

File: app/src/main/java/com/gxd/design/pattern/behavioral/chain/interceptor/impl/RetryAndFollowUpInterceptor.kt
package com.gxd.design.pattern.behavioral.chain.interceptor.impl

import com.gxd.design.pattern.behavioral.chain.interceptor.Chain
import com.gxd.design.pattern.behavioral.chain.interceptor.Interceptor
import com.gxd.design.pattern.behavioral.chain.interceptor.Response

class RetryAndFollowUpInterceptor : Interceptor {
    override fun intercept(chain: Chain): Response {
        val request = (chain as ChainImpl).request
        return chain.process(request)
    }
}
